src/__tests__/indexing.ts

```diff
@@ -0,0 +1,292 @@
+/**
+ * @jest-environment ./build/index.js
+ */
+import { jest } from '@jest/globals'
+import { CeramicApi, Page, StreamState, StreamUtils, TestUtils } from '@ceramicnetwork/common'
+import { ModelInstanceDocument } from '@ceramicnetwork/stream-model-instance'
+import { StreamID } from '@ceramicnetwork/streamid'
+import { config } from 'node-config-ts'
+import { createDid } from '../utils.js'
+import { DID } from 'dids'
+
+const TEST_MODEL = StreamID.fromString(config.jest.models[0])
+
+declare global {
+  const ceramic: CeramicApi
+  const ceramicClient: CeramicApi
+}
+
+const extractStreamStates = (page: Page<StreamState | null>): Array<StreamState> => {
+  if (page.edges.find(edge => edge.node === null)) {
+    throw new Error(
+      'null stream state found. This may indicate a problem with data persistence of your state store, which can result in data loss.'
+    )
+  }
+
+  return page.edges.map(edge => edge.node) as Array<StreamState>
+}
+
+const extractDocuments = (
+  ceramic: CeramicApi,
+  page: Page<StreamState | null>
+): Array<ModelInstanceDocument> => {
+  return extractStreamStates(page).map(state =>
+    ceramic.buildStreamFromState<ModelInstanceDocument>(state)
+  )
+}
+
+describe('indexing', () => {
+  describe('Using existing model', () => {
+    jest.setTimeout(1000 * 60)
+    const originalDid = ceramic.did as DID
+
+    const singleNodeTestCases: any[] = []
+    const twoNodesTestCases: any[] = []
+
+    if (config.jest.services.ceramic.indexingEnabled) {
```

Comment:
why do we need this?  Shouldn't indexing be enabled for all the nodes we want to run against?